### PR TITLE
[CTS]: New resource `opentelekomcloud_cts_event_notification_v3`

### LINKED ISSUE #2002: [CTS]: Implement event notifications module
CTS `event notification` API was released and need to be implemented.
Related documentation: https://docs.otc.t-systems.com/en-us/api/cts/cts_api_0312.html

### PR DESCRIPTION
**Pull request by artem-lifshits**
## Summary of the Pull Request
Implementation of new resource: `opentelekomcloud_cts_event_notification_v3`.

## PR Checklist

* [x] Closes: #2002
* [x] Refers to: #1948
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.
* [x] Release notes added.

## Acceptance Steps Performed

```
=== RUN   TestAccCTSEventNotificationV3_basic
--- PASS: TestAccCTSEventNotificationV3_basic (26.99s)
=== RUN   TestAccCTSEventNotificationV3_users
--- PASS: TestAccCTSEventNotificationV3_users (26.51s)
=== RUN   TestAccCTSEventNotificationV3_importBasic
--- PASS: TestAccCTSEventNotificationV3_importBasic (18.64s)
=== RUN   TestAccCheckCTSV3Validation
--- PASS: TestAccCheckCTSV3Validation (2.13s)
PASS

Process finished with the exit code 0
```

### Related documentation
https://docs.otc.t-systems.com/en-us/api/cts/cts_api_0312.html

### CODE REVIEW

**Review by Aloento — DISMISSED** [review]

File: docs/resources/cts_event_notification_v3.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+# opentelekomcloud_cts_event_notification_v3
+
+Allows to send SMS, email, or HTTP/HTTPS notifications through pre-configured SMN topics to subscribers.
+
+## Example Usage
+
+### Event notification which delivers every tenant action to subscribers
+
+```hcl
```
Comment (anton-sidelnikov) [inline docs/resources/cts_event_notification_v3.md:13]: add example with disabled topic

File: opentelekomcloud/services/cts/resource_opentelekomcloud_cts_event_notification_v3.go [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+	}
+
+	ctsN, err := keyevent.Create(client, createOpts)
+	if err != nil {
+		return fmterr.Errorf("error creating CTS event notification: %w", err)
+	}
+
+	d.SetId(buildEventNotificationID(ctsN.NotificationId, ctsN.NotificationName))
+
+	if status, ok := d.GetOk("status"); ok && createOpts.TopicId != "" {
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/cts/resource_opentelekomcloud_cts_event_notification_v3.go:198]: you really need this if, impossible to use one clause?

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

**Comment by artem-lifshits** [comment]
=== RUN   TestAccCTSEventNotificationV3_basic
--- PASS: TestAccCTSEventNotificationV3_basic (26.95s)
=== RUN   TestAccCTSEventNotificationV3_users
--- PASS: TestAccCTSEventNotificationV3_users (28.33s)
=== RUN   TestAccCTSEventNotificationV3_importBasic
--- PASS: TestAccCTSEventNotificationV3_importBasic (18.34s)
=== RUN   TestAccCheckCTSV3Validation
--- PASS: TestAccCheckCTSV3Validation (2.16s)
PASS

Process finished with the exit code 0

File: opentelekomcloud/services/cts/resource_opentelekomcloud_cts_event_notification_v3.go (hunk 2 above)
Comment (artem-lifshits) [reply opentelekomcloud/services/cts/resource_opentelekomcloud_cts_event_notification_v3.go:198]: fixed

File: docs/resources/cts_event_notification_v3.md (hunk 1 above)
Comment (artem-lifshits) [reply docs/resources/cts_event_notification_v3.md:13]: example added

**Review by Aloento — DISMISSED** [review]

File: opentelekomcloud/services/cts/resource_opentelekomcloud_cts_event_notification_v3.go (hunk 2 above)
Comment (artem-lifshits) [reply opentelekomcloud/services/cts/resource_opentelekomcloud_cts_event_notification_v3.go:198]: It's possible but imo this way is more understandable than something like this:
`if status, ok := d.GetOk("status"); ok && createOpts.TopicId != "" && status.(string) == "disabled" {`
What do you think?

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by Aloento — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**